pulser-core/pulser/channels/comparison_tools.py

```diff
@@ -0,0 +1,201 @@
+            and np.isclose(obj_dict[key], best_obj_dict[key], rtol=1e-14)
+        ):
+            comparison_ops.pop(key)
+    is_wrong_effective_ch = _compare_with_None(
+        comparison_ops, obj_dict, best_obj_dict
+    )
+    # Validates if no True in the dictionary of the comparisons
+    if not (True in is_wrong_effective_ch.values()):
+        return True
+    is_wrong_effective_index = list(is_wrong_effective_ch.values()).index(True)
+    is_wrong_key = list(is_wrong_effective_ch.keys())[is_wrong_effective_index]
+    raise ValueError(
+        f"{is_wrong_key} cannot be"
+        + (
+            " below "
+            if comparison_ops[is_wrong_key] == lt
+            else (
+                " above "
+                if comparison_ops[is_wrong_key] == gt
+                else " different than "
+            )
+        )
+        + f"{best_obj_dict[is_wrong_key]}."
+    )
+
+
+def validate_channel_from_best(
+    channel: Channel, best_channel: Channel
+) -> bool:
+    """Checks that a channel can be realized from another one.
+
+    Attributes:
+        channel: The channel to check.
+        best_channel: The channel that should have better properties.
+    """
+    if type(channel) != type(best_channel):
+        raise ValueError(
+            "Channels do not have the same types, "
+            f"{type(channel)} and {type(best_channel)}"
+        )
+    if channel.eom_config:
+        if best_channel.eom_config:
+            validate_eom_from_best(channel.eom_config, best_channel.eom_config)
+        else:
+            raise ValueError(
+                "eom_config cannot be defined in channel as the best_channel"
+                " does not have one."
+            )
+    best_ch_att = asdict(best_channel)
+    ch_att = asdict(channel)
+
+    # Error if attributes in channel and best_channel compare to True
+    comparison_ops = {
+        "addressing": ne,
+        "max_abs_detuning": gt,
+        "max_amp": gt,
+        "min_retarget_interval": lt,
+        "fixed_retarget_t": lt,
+        "max_targets": gt,
+        "clock_period": lt,
+        "min_duration": lt,
+        "max_duration": gt,
+        "mod_bandwidth": gt,
+    }
+    return _validate_obj_from_best(ch_att, best_ch_att, comparison_ops)
+
+
+def validate_eom_from_best(eom: BaseEOM, best_eom: BaseEOM) -> bool:
+    """Checks that an EOM config can be realized from another one.
+
+    Attributes:
+        eom: The EOM config to check.
+        best_eom: The EOM config that should have better properties.
+    """
+    best_eom_att = asdict(best_eom)
+    eom_att = asdict(eom)
+
+    # Error if attributes in eom and best_eom compare to True
+    comparison_ops = {"mod_bandwidth": gt}
+    if isinstance(eom, RydbergEOM):
+        if isinstance(best_eom, RydbergEOM):
+            comparison_ops.update(
+                {
+                    "limiting_beam": ne,
+                    "max_limiting_amp": gt,
```

Comment:
`max_limiting_amp` needs to be `ne` too because it influences the lightshift (potentially...)